src/crimson/osd/recovery_backend.cc

```diff
@@ -55,15 +55,12 @@ void RecoveryBackend::clean_up(ceph::os::Transaction& t,
   replica_push_targets.clear();
 
   for (auto& [soid, recovery_waiter] : recovering) {
-    if ((recovery_waiter->pull_info
-         && recovery_waiter->pull_info->is_complete())
-	|| (!recovery_waiter->pull_info
-	  && recovery_waiter->obc && recovery_waiter->obc->obs.exists)) {
+    if (recovery_waiter->obc) {
```

Comment:
Shouldn't we also check for `&& recovery_waiter->obc->obs.exists` here?

Reply:
I don't think that's relevant -- if it has a recovery_waiter, it needs to be interrupted.

Reply:
We will interrupt the recovery_waiter in line 63. I was referring to whether we should try to interrupt a non-existing (yet to be created) obc.

Reply:
Um, I think `!obs->exists` obc may also be locked when it's being loaded from disk, which means it also needs to be interrupted, am I right?